src/function/table/arrow_conversion.cpp

```diff
@@ -18,24 +18,34 @@ static void ShiftRight(unsigned char *ar, int size, int shift) {
 	}
 }
 
+idx_t GetEffectiveOffset(ArrowArray &array, int64_t parent_offset, ArrowScanLocalState &state,
+                         int64_t nested_offset = -1) {
+	if (nested_offset != -1) {
+		// The parent of this array is a list
+		// We just ignore the parent offset, it's already applied to the list
+		return array.offset + nested_offset;
+	}
+	// Parent offset is set in the case of a struct, it applies to all child arrays
+	// 'chunk_offset' is how much of the chunk we've already scanned, in case the chunk size exceeds
+	// STANDARD_VECTOR_SIZE
+	return array.offset + parent_offset + state.chunk_offset;
+}
+
```

Comment:
Should we have an offset struct that holds all the offset information and have this method in?
Then we can localize this offset logic and not pass two values all over.

Reply:
We would still be populating this on the fly, the nested offset/parent offset is provided by the parent array.

I was thinking of writing this offset into the `ArrowArrayScanState` every time we scan a new chunk, so we don't repeat this logic everywhere, but the only thing we can get rid of that way is the parent_offset/nested offset if/else.

We would still need to keep the context of what type of offset this is though, because the `state.chunk_offset` would still need to be applied for every scanned Vector.